tests/lean/run/coinductive_syntax.lean

```diff
@@ -103,3 +103,7 @@ end
 coinductive my_nat  where
   | zero : my_nat
   | succ : my_nat → my_nat
+
+def Set := Nat → Prop
+
+coinductive Foo : Set where
```

Comment:
Maybe also add constructors to this test, to see if the rest of the machinery (constructing constructors) can handle this particular pitfall well :)